docs/pages/developer-guide.md

```diff
@@ -936,6 +935,174 @@ In the example above, a server task can be accessed through the address:
+```
+DNSName([discovery-prefix-index].[service-name].autoip.dcos.thisdcos.directory)
+```
+
+This certificate configuration allows to run proper TLS hostname verification by client when task is accessed by one of [internal DNS names](https://docs.mesosphere.com/1.9/networking/dns-overview/).
+
+Each certificate is valid for **10 years**.
+
+### Artifacts format
+
+TLS artifacts can be provided to a task in two different formats:
+
+#### PEM
+
+Standard [PEM encoded](https://en.wikipedia.org/wiki/Privacy-enhanced_Electronic_Mail) certificate, private key and root CA certificate. Based on a file extension name it is possible to tell which artifact is in the file.
+
+A certificate file with **.crt** extension contains an end-entity certificate with possible chain of certificates leading to the root CA certificate. The root CA certificate is not part of end-entity certificate file.
+
+A **.key** file contains PEM encoded private key.
+
+A file with **.crt** extension contains root CA certificate without certificate chain.
+
+#### Java Keystore
+
+The [Java Keystore (JKS)](https://en.wikipedia.org/wiki/Keystore) is a repository of various certificates and private keys. When private key and certificate is requested a task will get 2 JKS format files.
+
+A **.keystore** is a JKS file that contains a private key with end-entity certificate and complete certificate chain including the root CA certificate. Certificate is stored under alias name **`default`**. The keystore and key are protected by single **`notsecure`** password.
+
+A **.truststore** is a JKS file that contains DC/OS root CA certificate stored as a trust certificate. The certificte is stored under alias **`dcos-root`**. The keystore is protected by **`notsecure`** password.
+
+The password **`notsecure`** that is protecting both JKS files (containing an end-entity certificate with private key and root CA certificate) is selected because most of Java tools and libraries validate for some password. It is not to meant to provide any additional protection. Security of both files is achieved by using DC/OS secrets store with file-based in-memory secrets. No other schedulers and tasks can access TLS artificates provisioned by a scheduler.
+
+**Imporant note:** The DC/OS 1.10 secrets store doesn't provide support for storing binary data. To overcome this limitation the `keystore` and `truststore` files are encoded using Base64 encoding when stored to secrets store. Files are mounted into a task container with additional `base64` file name suffix - `*.keystore.base64` and `*.truststore.base64`. There is a `bootstrap` utility helper that automatically decodes files in `$MESOS_SANDBOX` with `*.keystore.base64` and `*.truststore.base64` extension and stores them in a sandbox directory with expected `*.keystore` and `*.truststore` names. **Decoded files aren't stored using in-memory file system and are stored on a disk even after task finishes until the sandbox is removed by an operator or Mesos GC.**
```

Comment:
To overcome this limitation, the `keystore` and `truststore` files are Base64-encoded before they are added to the secrets store.